code/modules/ai/effect_node.dm

```diff
@@ -0,0 +1,60 @@
+//The node with a physical presence on the map; mainly exists as a actual object to be able to be highlighted easily for easy debugging and viewing purposes
+//It also holds the datum ai_node, containing getter procs and handle info like weights
+
+/obj/effect/ai_node //A effect that has a ai_node datum in it, used by AIs to pathfind over long distances as well as knowing what's happening at it
+	name = "AI Node"
+	icon = 'icons/effects/landmarks_static.dmi'
+	icon_state = "6x" //Pure white 'X' with black borders
+	var/datum/ai_node/datumnode = new/datum/ai_node() //Stores things about the AI node
+	var/turf/srcturf //The turf this is on
+	anchored = TRUE //No pulling those nodes yo
+	alpha = 0 //Not visible at all
+
+/obj/effect/ai_node/Initialize() //Add ourselve to the global list of nodes
+	. = ..()
+	srcturf = loc
+	datumnode.parentnode = src
+	GLOB.allnodes += src
+
+//Current weights are nothing, wait for more ai updates
+//Parameter call example
+//GetBestAdjNode(list(ENEMY_PRESENCE = -100, DANGER_SCALE = -1)) if these were defined
+//This means that the proc will pick out the *best* node
+
+/obj/effect/ai_node/proc/GetBestAdjNode(list/weight_modifiers)
+	if(weight_modifiers)
+		var/obj/effect/ai_node/node_to_return = datumnode.adjacent_nodes[1]
+		var/current_best_node = 0
+		var/current_score = 0
+		for(var/obj/effect/ai_node/node in shuffle(datumnode.adjacent_nodes)) //We keep a score for the nodes and see which one is best
+			for(var/i = 1 to length(weight_modifiers))
+				current_score += (node.datumnode.get_weight(i) * weight_modifiers[i])
+
+			if(current_score >= current_best_node)
+				current_best_node = current_score
+				node_to_return = node
+			current_score = 0
```

Comment:
Just define `current_score` inside the `for` loop, since it's not used elsewhere?